source/server/listener_manager_impl.cc

```diff
@@ -1049,6 +1049,17 @@ void ListenerManagerImpl::maybeCloseSocketsForListener(ListenerImpl& listener) {
     // close the socket because they need to receive packets for existing connections via the
     // listen sockets.
     listener.listenSocketFactory().closeAllSockets();
+
+    // In case of this listener was in-place updated previously and in the filter chains draining
+    // procedure, so close the sockets for the previous draining listener.
+    auto iter = draining_filter_chains_manager_.begin();
+    while (iter != draining_filter_chains_manager_.end()) {
```

Comment:
nit: I think you can just use a normal for each loop here without iterators?

Reply:
Yea, sure, thanks